### PR TITLE
[BACKLOG-22048] Foundry CT2000 Dashboard

### PR DESCRIPTION
**Pull request by davidmsantos90**
@webdetails/millenniumfalcon please review.

### CODE REVIEW

File: core-js/src/main/javascript/cdf/queries/SolrQuery.js [hunk 1]
```diff
@@ -349,6 +320,102 @@ define([
     // endregion
 
     // region Private Methods
+    __exportDataSuccess: function(options, data) {
+      var isBlobAvailable = Blob != null;
```
Comment (dcleao) [inline core-js/src/main/javascript/cdf/queries/SolrQuery.js]: Shouldn't this be `typeof Blob !== "undefined"` ?

File: core-js/src/main/javascript/cdf/queries/SolrQuery.js [hunk 2]
```diff
@@ -116,179 +118,148 @@ define([
...
+     * of the parameters macros used in Solr Query Request.
+     *
+     * @return {object} The macros' values in the request.
+     */
+    getParameterMacros: function() {
+      var parameterMacros = {};
 
-      this.setOption("url", this.__buildSolrUrl());
-    },
+      var macros = this.getOption("macros");
```
Comment (dcleao) [inline core-js/src/main/javascript/cdf/queries/SolrQuery.js]: I don't understand where this `macros` array comes from. Shouldn't this be the result of analysis of the several specified data source property values?

File: core-js/src/main/javascript/cdf/queries/SolrQuery.js [hunk 3]
```diff
@@ -123,22 +123,27 @@ define([
...
-      macros.forEach(function(macro) {
-        var name = macro[0];
-        var value = this.__getDashboardParameterValue(name, macro[1]);
+      for (var name in macros) {
+        if (macros.hasOwnProperty(name)) {
+          var value = this.__getDashboardParameterValue(name, macros[name]);
 
-        if (Utils.isString(value)) value = value.split(/,\s*/);
+          if (Utils.normalizeValue(value) !== null) {
+            if (Utils.isString(value)) value = value.split(/,\s*/);
```
Comment (dcleao) [inline core-js/src/main/javascript/cdf/queries/SolrQuery.js]: Remember `/\s*,\s*/` ?

File: core-js/src/test/javascript/cdf/queries/SolrQuery-spec.js [hunk 4]
```diff
@@ -19,7 +19,7 @@ define([
 
   /* globals beforeEach, describe, it, expect, spyOn */
 
-  describe("SolrQuery", function () {
+  fdescribe("SolrQuery", function () {
```
Comment (dcleao) [inline core-js/src/test/javascript/cdf/queries/SolrQuery-spec.js]: `fdescribe` :-D

**Comment by davidmsantos90** [comment]
@webdetails/millenniumfalcon please review

**Pull request merged**